Assets/MirageSDK/Plugins/WalletConnect.VersionShared/Utils/WalletDownloadHelper.cs

```diff
@@ -84,21 +84,32 @@ public static async UniTask<WalletEntry> FindWalletEntry(Wallets wallet, bool in
 					return supportedWallets;
 				}
 
-				var filteredSupportedWallets = GetAllSupportedWallets(walletconnectSupportedWallets: supportedWallets);
-				foreach (var wallet in filteredSupportedWallets.Values)
+				var filteredSupportedWallets = GetAllSupportedWallets(walletConnectSupportedWallets: supportedWallets);
+
+				if (filteredSupportedWallets != null)
 				{
-					await wallet.DownloadImages();
+					foreach (var wallet in filteredSupportedWallets.Values)
+					{
+						await wallet.DownloadImages();
```

Comment:
can we construct a list of tasks and use UniTask.WhenAll to avoid waiting for the images one by one?